cmd/crates/soroban-test/tests/fixtures/test-wasms/token/src/test.rs

```diff
@@ -14,183 +13,154 @@ fn create_token<'a>(e: &Env, admin: &Address) -> TokenClient<'a> {
     token
 }
 
+fn auth_invoke(
+    e: &Env,
+    token: &TokenClient,
+    fn_name: &str,
+    args: soroban_sdk::Vec<Val>,
+) -> AuthorizedInvocation {
+    AuthorizedInvocation {
+        function: AuthorizedFunction::Contract((token.address.clone(), fn_name.into_val(e), args)),
+        sub_invocations: std::vec![],
+    }
+}
+
 #[test]
 fn test() {
-    let e = Env::default();
+    let e = &Env::default();
     e.mock_all_auths();
 
-    let admin1 = Address::generate(&e);
-    let admin2 = Address::generate(&e);
-    let user1 = Address::generate(&e);
-    let user2 = Address::generate(&e);
-    let user3 = Address::generate(&e);
-    let token = create_token(&e, &admin1);
+    let admin1 = Address::generate(e);
+    let admin2 = Address::generate(e);
+    let user1 = &Address::generate(e);
+    let user2 = &Address::generate(e);
+    let user3 = &Address::generate(e);
+    let token = &create_token(e, &admin1);
 
-    token.mint(&user1, &1000);
+    token.mint(user1, &1000);
     assert_eq!(
         e.auths(),
         std::vec![(
             admin1.clone(),
-            AuthorizedInvocation {
-                function: AuthorizedFunction::Contract((
-                    token.address.clone(),
-                    symbol_short!("mint"),
-                    (&user1, 1000_i128).into_val(&e),
-                )),
-                sub_invocations: std::vec![]
-            }
+            auth_invoke(e, token, "mint", (user1, 1000_i128).into_val(e))
```

Comment:
I think this code would be clearer if you kept the type and just added a `#[allow(clippy::...)]`.

It feels rather arbitrary to move this block out into a function, especially on an arbitrary number of lines.